src/stan/model/indexing/assign.hpp

```diff
@@ -2,6 +2,7 @@
 #define STAN_MODEL_INDEXING_ASSIGN_HPP
 
 #include <stan/math/prim.hpp>
+#include <stan/model/indexing/assign_varmat.hpp>
```

Comment:
I would prefer if you moved the `assign_impl` overloads that are not varmat specific into this file, so there was no need for this include.